docs/src/doc/docs/developer_guide/architecture/architecture_overview.md

```diff
@@ -0,0 +1,106 @@
+# Architecture overview
+
+Normally, you would think that a tool like `Dokka` simply parses some programming language sources and generates
+`HTML` pages for whatever it sees along the way, with little to no abstractions. That would be the simplest and
+shortest way to implement a documentation engine.
+
+However, it was clear that `Dokka` may need to generate documentation from various sources (not only `Kotlin`), that users
+might request additional output formats (like `Markdown`), that users might need additional features like supporting
+custom `KDoc` tags or rendering `mermaid.js` diagrams - all these things would require changing a lot of code inside
+`Dokka` itself if all solutions were hardcoded.
+
+For this reason, `Dokka` was built from the ground up to be extensible and customizable. You can think of the general
+flow of generating documentation with `Dokka` as mapping one intermediate representation / abstraction into another.
+You, as a `Dokka` developer or a plugin writer, can use extension points and introduce selective changes to the
+model on one particular level without touching the rest.
+
+## Overview of data model
+
+Below you can find the general flow and transformation of Dokka's models.
+
+```mermaid
+flowchart TD
+    Input --> Documentables --> Documentables --> Pages --> Pages --> Output
```

Comment:
As for me all arrows should have explanations.

`flow` is misleading. Do you mean data flow, control flow or something else?!
Maybe `pipeline`  would be better.

Reply:
Seems like loops are not necessary for this particular page, since it's an overview of the data model. The fact that you can transform documentables relates more to extension points, so I'll remove the loops from this chart.

I've changed it to use `pipeline` word and added some more examples and explanations, hopefully it's better now